File: Android/doraemonkit-weex/src/main/java/com/didichuxing/doraemonkit/weex/common/DKCommonActivity.java
package com.didichuxing.doraemonkit.weex.common;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;

import com.didichuxing.doraemonkit.kit.core.BaseActivity;
import com.didichuxing.doraemonkit.kit.core.BaseFragment;

/**
 * @author haojianglong
 * @date 2019-06-18
 */
public class DKCommonActivity extends BaseActivity {

    private static final String CLASSNAME = "className";

    public static void startWith(Context context, Class<? extends BaseFragment> clazz) {
        Intent intent = new Intent(context, DKCommonActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        intent.putExtra(CLASSNAME, clazz);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle bundle = getIntent().getExtras();
        if (bundle == null) {
            finish();
            return;
        }
        Class<BaseFragment> clazz = (Class<BaseFragment>) getIntent().getSerializableExtra(CLASSNAME);
        showContent(clazz);
    }

}

File: Android/kotlin/doraemonkit-weex/src/main/java/com/didichuxing/doraemonkit/weex/common/DKCommonActivity.kt
package com.didichuxing.doraemonkit.weex.common

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.didichuxing.doraemonkit.kit.core.BaseActivity
import com.didichuxing.doraemonkit.kit.core.BaseFragment

/**
 * Transformed by alvince on 2020/6/30
 *
 * @author haojianglong
 * @date 2019-06-18
 */
class DKCommonActivity : BaseActivity() {

    companion object {
        private const val CLASSNAME = "className"

        @JvmStatic
        fun startWith(context: Context, clazz: Class<out BaseFragment>) {
            val intent = Intent(context, DKCommonActivity::class.java)
            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
            intent.putExtra(CLASSNAME, clazz)
            context.startActivity(intent)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        intent.extras
            ?: run {
                finish()
                return
            }

        (intent.getSerializableExtra(CLASSNAME) as? Class<BaseFragment>)
            ?.also { showContent(it) }
    }

}